### PR TITLE
doc: list dep. versions / Vagrant improvements

### PR DESCRIPTION
**Pull request by ianvernon**
* Move contents of vagrant.rst, which was previously not accessible in our documentation, to contributing.rst so all information about the development Vagrant VMs is in one location.
* List required versions of build / developer dependencies in a singular location that can be referenced throughout the docs.

Signed-off by: Ian Vernon <ian@covalent.io>

### CODE REVIEW

**Review by tgraf — APPROVED** [review]
Looks great, I'll wait for @scanf's review before merging.

**Review by aalemayhu — APPROVED** [review]
PDF and HTML pages looks good. Thanks.

**Pull request merged**
